### PR TITLE
Bcrypt

### PR DESCRIPTION
**Pull request by sjaeckel**
~Add the bcrypt password-based hashing algorithm.~

Add the bcrypt PBKDF algorithm as used when encrypting OpenSSH key files.

### Checklist

* [x] documentation is added or updated
* [x] tests are added or updated

### CODE REVIEW

File: src/headers/tomcrypt_misc.h [hunk 1]
```diff
@@ -59,6 +59,13 @@ int base16_decode(const          char *in,  unsigned long  inlen,
                         unsigned char *out, unsigned long *outlen);
 #endif
 
+#ifdef LTC_BCRYPT
+int bcrypt(const          char *password, unsigned long password_len,
```
Comment (karel-m) [inline src/headers/tomcrypt_misc.h]: Providing the fact that we have test cases like `"pass\0word"` I would rather use `const unsigned char *password` or even `const unsigned char *secret`

Comment (sjaeckel) [reply src/headers/tomcrypt_misc.h]: IIUC "pass\0word" is no valid password for most password-based crypto applications (e.g. SCRAM) but it's no problem for us to handle it, so how about changing it to `const void*`?

Another point I realized in the last days... would it probably make sense to rename this function to `bcrypt_openssh()` as it's a specific version which differs from the original brcypt proposal?

Comment (karel-m) [reply src/headers/tomcrypt_misc.h]: The `const void*` is definitely better than `const char*` and AFAIK we already use it like this somewhere (pkcs8?).

Did you mean `bcrypt_openssh` or `bcrypt_openbsd`?

Comment (sjaeckel) [reply src/headers/tomcrypt_misc.h]: > `bcrypt_openssh` or `bcrypt_openbsd`

I don't have an opinion :-) maybe `bcrypt_openbsd` is better as it was first used there.

**Pull request merged**